### PR TITLE
infra: add PL device testing using a stable PL version

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*
Tests the latest Braket code against the latest release of PennyLane. Includes a tox env for better local testing.

*Testing done:*
`tox -e pennylane-device-tests`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-pennylane-plugin-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/braket-latest-pl-stable.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          echo 'Error: Version of PennyLane does not provide device integration tests.' >&2
+        else
+          pl-device-test --device=braket.local.qubit --tb=short --skip-ops --shots=20000 -k 'not no_0_shots'
+          pl-device-test --device=braket.local.qubit --tb=short --skip-ops -k 'not Sample and not no_0_shots'
+        fi
+    - name: Upload coverage report to Codecov
+      uses: codecov/codecov-action@v4
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+      if: ${{ strategy.job-index }} == 0
```
Comment (speller26) [inline .github/workflows/braket-latest-pl-stable.yml:59]: Don't think we need codecov for this one

**Review by speller26 — COMMENTED** [review]
This should this be `infra`; it adds a workflow, and the only change to `setup.py` is in `extras_require["test"]`

**Review by speller26 — COMMENTED** [review]
Is there a way to add this as a tox target so we can run the device tests locally?

**Comment by math411** [comment]
> Is there a way to add this as a tox target so we can run the device tests locally?

@speller26, added `tox -e pennylane-device-tests`

**Review by speller26 — APPROVED** [review]

**Pull request merged**